extension/js/common/core/crypto/smime/smime-key.ts

```diff
@@ -1,18 +1,23 @@
 /* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
-import * as forge from 'node-forge';
 import { Key, UnexpectedKeyTypeError } from '../key.js';
 import { Str } from '../../common.js';
 import { UnreportableError } from '../../../platform/catch.js';
 import { PgpArmor } from '../pgp/pgp-armor.js';
 import { Buf } from '../../buf.js';
 import { MsgBlockParser } from '../../msg-block-parser.js';
 import { MsgBlock } from '../../msg-block.js';
-
+import * as forge from 'node-forge';
 export type SmimeMsg = forge.pkcs7.PkcsEnvelopedData;
 export const DATA_OID = '1.2.840.113549.1.7.1';
 export const SIGNED_DATA_OID = '1.2.840.113549.1.7.2';
 export const ENVELOPED_DATA_OID = '1.2.840.113549.1.7.3';
 export class SmimeKey {
+  private static get forgeCustom(): typeof forge {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return forge.default ? forge.default : forge;
+  }
```

Comment:
@sosnovsky How about now? What do you think about solution?
When I import like above `import * as forge` then i have to call `forge.default.asn1` instead of `forge.asn1`. Therefore I added above logic.
Maybe there could be simple solution for this.

Let me know your suggestion.

Reply:
Ok, I'm currently fixing `test re-auth after updating chrome extension` test (It fails because OAuth2.js coudln't get window object because it was running from background). Fixing it to pass requried screen dimensions to service worker and pass it to OAuth2.

Let me have a look at  `ajaxGmailAttachmentGetChunk` after current one

Reply:
I have a question. Why don't we just use `fetch` in `ajaxGmailAttachmentGetChunk`?

Reply:
Also I think we couldn't move `ajaxGmailAttachmentGetChunk` to content script because of below reason?
https://github.com/FlowCrypt/flowcrypt-browser/blob/cb1423a3bcbeecb3a8471c457c94a8e76280d9fd/extension/js/common/api/email-provider/gmail/gmail.ts#L160

Reply:
Okay, Got it.

Reply:
Hmm.. This manifest v3 migration really has tons of problems.

`URL.createObjectURL` is now removed in service worker and therefore we can't use it.

Currently finding solution now.

https://github.com/FlowCrypt/flowcrypt-browser/blob/a852f8779ff0d921e49fa5d79b07196d62abdd65/extension/js/common/browser/browser-msg.ts#L556
https://github.com/FlowCrypt/flowcrypt-browser/blob/4edbf1836cff75b6262bacc1757f45382661520c/extension/js/common/browser/browser.ts#L16
https://groups.google.com/a/chromium.org/g/chromium-extensions/c/u0NH7L3v9L4?pli=1#:~:text=Some%20workarounds%3A,use%C2%A0URL.createObjectURL.

Reply:
Do you agree using offscreen api?

Reply:
> Do you agree using offscreen api?

Yes, let's use it.

> Hmm.. This manifest v3 migration really has tons of problems.

Yep, that's why we deferred this migration, but now there is no choice as manifest v2 will be fully deprecated this summer :(

Reply:
@sosnovsky I got a bit confused. We tried to create object url (URL.createObjectURL) because browser message can't send a lot of data. But if we use offscreen document we also need to send message to offscreen blob data.
Isn't this what we wanted to avoid with createObjectURL?

Please correct me if I'm wrong.

https://github.com/FlowCrypt/flowcrypt-browser/blob/a852f8779ff0d921e49fa5d79b07196d62abdd65/extension/js/common/browser/browser-msg.ts#L556
https://github.com/FlowCrypt/flowcrypt-browser/blob/a852f8779ff0d921e49fa5d79b07196d62abdd65/extension/js/common/browser/browser-msg.ts#L545

Reply:
It seems there is no strict limit on maximum size of browser message - it'll probably depend on available system memory. Here we'll have maximum size of 10-15mb (as it's the max for message attachments), so let's try to remove usages of `URL.createObjectURL` and send original objects instead.

Reply:
Ok

Reply:
Ok

Reply:
I removed `URL.createObjectURL` usage but `decrypt - public key is rendered minimized for outgoing messages` test is failing because of that. I tried to run this test in master branch with `URL.createObjectURL` usage removed and it also failed.
I'm digging more.

Reply:
Are there any errors in console? I think public key size shouldn't be too large for browser msg

Reply:
No errors in console

Reply:
No errors in console

Reply:
I found reason and it's because when message is returned from `ajaxGmailGetChunk`, with the `createObjectUrl` usage removed `object` is returned instead of `Buf`.
This is a temp fix and it fixed issue.
https://github.com/FlowCrypt/flowcrypt-browser/pull/5607/commits/4262aeeed49e11f09caf0a31413b7515a63c28b7

Reply:
I found reason and it's because when message is returned from `ajaxGmailGetChunk`, with the `createObjectUrl` usage removed `object` is returned instead of `Buf`.
This is a temp fix and it fixed issue.
https://github.com/FlowCrypt/flowcrypt-browser/pull/5607/commits/4262aeeed49e11f09caf0a31413b7515a63c28b7

Reply:
Also FYI created new PR as we also have to test live test with new branch name

Reply:
> I found reason and it's because when message is returned from `ajaxGmailGetChunk`, with the `createObjectUrl` usage removed `object` is returned instead of `Buf`. This is a temp fix and it fixed issue. [4262aee](https://github.com/FlowCrypt/flowcrypt-browser/commit/4262aeeed49e11f09caf0a31413b7515a63c28b7)

Ok, let's just add `// todo: it's a temporary fix, should be removed` comment, so we won't forget about it

Reply:
Okay